plugin/modify-env/modify-env.c

```diff
@@ -113,14 +113,20 @@ char * read_dmtcp_env_file(char *file, int size) {
   }
 #else
   char pathname[512];
-  if (strlen(dmtcp_get_ckpt_dir()) > sizeof(pathname)-1-sizeof(file)) {
-    warning(__FILE__ ": Pathname of ckpt dir is too long: ",
-            dmtcp_get_ckpt_dir() /* , "\n" */ );
-    exit(1);
+  // Fix for picky compiler.  The function dmtcp_get_ckpt_dir() would only
+  //   return NULL if we were not running under DMTCP.  But the compiler
+  //   doesn't know that.
+  const char *dmtcp_ckpt_dir = dmtcp_get_ckpt_dir();
+  if (dmtcp_ckpt_dir) {
+    if (strlen(dmtcp_ckpt_dir) > sizeof(pathname)-1-sizeof(file)) {
```

Comment:
Are you sure it should be `sizeof(file)` here? It seems like the variable `file` is a pointer.